velox/expression/CastExpr-inl.h

```diff
@@ -206,7 +241,9 @@ void CastExpr::applyCastKernel(
     if constexpr (
         TypeTraits<ToKind>::isPrimitiveType &&
         TypeTraits<ToKind>::isFixedWidth) {
-      if (inputRowValue.size() == 0) {
+      if (inputRowValue.size() == 0 ||
+          (!Truncate && isIntegralOrBoolean(ToKind) &&
```

Comment:
Wondering if it make sense to change isIntegralOrBoolean to compile. time check using std::is_same_v.

Reply:
@mbasmanova , Masha, great to know we can do compile time check here. I came across std::is_integral_v<> when looking up std::is_same_v<>, so I can use it here for simplicity.